src/app/shared/view-issue/tester-response/tester-response.component.ts

```diff
@@ -190,28 +194,20 @@ export class TesterResponseComponent implements OnInit, OnChanges {
     }
 
     const updatedIssue = this.issue.clone(this.phaseService.currentPhase);
-    const values = this.testerResponseForm.getRawValue();
-    const disagrees = [];
-    const reasons = [];
-
-    let index = 0;
-    for (const [key, value] of Object.entries(values)) {
-      if (key.startsWith('disagree-box')) {
-        disagrees.push(value);
-      } else if (key.startsWith('tester-response')) {
-        reasons.push(value);
-      }
-      index++;
-    }
+    const values: {} = this.testerResponseForm.getRawValue();
+
+    updatedIssue.testerResponses.map((response: TesterResponse, index: number) => {
+      // Filter Keys based on Response Index
+      const responseDataKeys = Object.keys(values).filter((key: string) => key.includes(`${index}`));
```

Comment:
Can we simplify this method, by creating a separate method that retrieves the `isDisagree` value and `reason for disagreement` associated with any specific response item in the FormGroup (given its index number)?

We can then simply use the `isDisagree` value and response here.
In addition, we could retrieve these directly from the form control, given its index? (i.e.  no need to use `getRawValue`)

example:
`const isDisagree = this.testerResponseForm.get(this.getDisagreeRadioFormId(index)).value;`

Reply:
Sure 👍